src/v/kafka/server/tests/create_partition_test.cc

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "config/configuration.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/server/tests/topic_properties_helpers.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "model/namespace.h"
+
+#include <seastar/core/loop.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/util/defer.hh>
+
+#include <absl/container/flat_hash_map.h>
+#include <boost/test/tools/context.hpp>
+
+using namespace std::chrono_literals; // NOLINT
+
+inline ss::logger test_log("test"); // NOLINT
+
+FIXTURE_TEST(
+  test_unlicensed_topic_prop_create_partition, topic_properties_test_fixture) {
+    using props_t = absl::flat_hash_map<ss::sstring, ss::sstring>;
+    using test_t = std::pair<std::string_view, props_t>;
+    const auto with = [](const std::string_view prop, const auto value) {
+        return std::make_pair(
+          prop, props_t{{ss::sstring{prop}, ssx::sformat("{}", value)}});
+    };
+
+    update_cluster_config(lconf().enable_schema_id_validation.name(), "compat");
+
+    std::initializer_list<test_t> enterprise_props{
```

Comment:
nitpick: I might instinctively include a non-failing case, but I suppose that's covered by extant tests. Probably nothing to do here.